hudi-client/src/main/java/org/apache/hudi/client/HoodieWriteClient.java

```diff
@@ -186,11 +186,15 @@ protected void rollBackInflightBootstrap() {
    * @return JavaRDD[WriteStatus] - RDD of WriteStatus to inspect errors and counts
    */
   public JavaRDD<WriteStatus> upsert(JavaRDD<HoodieRecord<T>> records, final String instantTime) {
+    return upsert(records, instantTime, null);
+  }
+
+  public JavaRDD<WriteStatus> upsert(JavaRDD<HoodieRecord<T>> records, final String instantTime, String schema) {
```

Comment:
would we use Option<String> schema here to avoid NPE?

Reply:
could we get schema string from HoodieWriteConfig?

Reply:
ok